File: exercice2_3.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/types.h>
#include <sys/wait.h>
#include <errno.h>

int main()
{
	pid_t pid_fils ;
	int i = 0 ;
	for (i = 1; i <= 5; ++i)
	{
		pid_fils = fork() ;
		if (pid_fils == -1)
		{
			perror("Problème dans fork lors de création de fils\n") ;
			exit (errno) ;
		}
		else if (pid_fils == 0)
		{
			//printf("je suis le fils %d\n", i) ;
			break ;
		}
	}
	for (i = 1; i <= 5; ++i)
	{
		wait(NULL) ;
	}
	return (0) ;
}

File: Python/exercice2_3.py
import os
import sys
import time

def main():
    
    for i in range(1, 6):
        pid_fils = os.fork()
        
        if pid_fils == -1:
            print(f"Problème dans fork lors de création de fils {i}")
            sys.exit(1)
        
        elif pid_fils == 0:
            print(f"je suis le fils {i}")
            return 
            
    for i in range(1, 6):
        os.wait()

if __name__ == "__main__":
    main()